EIPS/eip-BN256_HashToCurve.md

```diff
@@ -0,0 +1,342 @@
+and pairing checks in BN256.
+Reducing these costs in
+[EIP-1108](./eip-1108.md)
+made [ETHDKG](https://github.com/PhilippSchindler/ethdkg),
+a distributed key generation protocol in Ethereum,
+less expensive.
+ETHDKG by itself is useful; however, what it is lacking is
+the ability to verify arbitrary BLS signatures.
+Creating group signatures by aggregating partial signatures
+is one goal of a DKG protocol.
+The DKG enables the computation of partial signatures to be
+combined into a group signature offline, but there is no
+easy way to verify partial signatures or group signatures
+in the EVM.
+
+In order to perform BLS signature validation, a hash-to-curve
+function is required.
+While the MapToGroup method initially discussed in the original BLS
+[paper](https://crypto.stanford.edu/~dabo/pubs/abstracts/weilsigs.html)
+works in practice, the nondeterministic nature of the algorithm
+leaves something to be desired, as we would like to bound
+the overall computational cost in the EVM.
+A deterministic method for mapping to BN curves is given
+[here](https://www.di.ens.fr/~fouque/pub/latincrypt12.pdf);
+in the paper, Fouque and Tibouchi proved their mapping
+was indifferentiable from a random oracle.
+This paper gives us the desired algorithm.
+
+## Specification
+We mostly follow the Fouque and Tibouchi
+[paper](https://www.di.ens.fr/~fouque/pub/latincrypt12.pdf)
+for the hash-to-curve algorithm.
+This algorithm, as with the current IRTF
+[draft](https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-10),
+separates the hash-to-curve operation into a
+hash-to-field function and a deterministic field-to-curve map.
+We include some improvements from another
+[paper](https://eprint.iacr.org/2019/403);
+one improvement enables there to be no special points in the
+deterministic mapping as long as the multiplicative inverse
+function call ensures `inverse(0) == 0`;
+this always happens when inversion is computed via exponentiation.
+Furthermore, the Wahby and Boneh paper allows us to make
+a hash-to-curve algorithm for mapping to G2;
+the algorithm is effectively the same, and the only difference
+in the different constants and the base field.
+All of the required operations in these functions
+involve operations on the underlying finite fields.
+Algorithms for necessary GF(p^2) operations are referenced below.
+
+On a local machine, ScalarMult was clocked at 68 microseconds
+per operation.
+The same machine clocked HashToG1 at 94 microseconds per operation
+when hashing 32 bytes into G1; it took 105 microseconds per operation
+when hashing 1024 bytes into G1.
+Given that it currently costs 6000 gas for ECMul, this gives us
+an estimated gas cost for HashToG1 at 8500 + len(bytes).
+Similarly, HashToG2 was clocked at 886 microseconds per operation
+when hashing 32 bytes into G2 and 912 microseconds per operation when
+hashing 1024 bytes into G2.
+This allows us to estimate the gas cost at 80000 + 3*len(bytes).
```

Comment:
The specification should be *purely* technical, completely ignoring all history, sources, etc.  A good specification section looks more like code than an essay and just describes the spec.  If you want to talk about where things come from, or why decisions were made then Motivation or Rationale section are more appropriate.  Same goes for most of this paragraph.

The Rationale section is where you should put your reasons for making certain decisions within the specification.  For example, why you chose `80000 + 3 * len(bytes)` is very appropriate for the Rationale section, while the specification section should just define a constant or a function with little to know reasoning behind it.

The thing to keep in mind is that someone who just wants to implement the spec and doesn't care about the "why" at all (e.g., their boss told them to go implement EIP-3068) should be able to skip to the Specification section and immediately get down to business.

Reply:
Same basic content throughout the Specification section.

Reply:
Commentary in Specification has been reduced.